libnavui-maneuver/src/test/java/com/mapbox/navigation/ui/maneuver/model/RoadShieldGeneratorTest.kt

```diff
@@ -93,29 +96,17 @@ class RoadShieldGeneratorTest {
         val mockShieldText = "880"
         val mockDesiredHeight = 50
         val mockResources = ctx.resources
-        val mockShieldIcon = byteArrayOf()
+        val emptyShieldIcon = byteArrayOf()
 
         val spannable = RoadShieldGenerator.styleAndGetRoadShield(
             mockShieldText,
             mockDesiredHeight,
-            mockResources
-        )
-        val imageSpan = spannable.getSpans(0, spannable.length, ImageSpan::class.java)
-
-        assertTrue(spannable.isNotEmpty())
-        assertTrue(imageSpan.isEmpty())
-    }
-
-    @Test
-    fun `when road shield invalid then spannable has empty image span`() {
-        val mockShieldText = "880"
-        val mockDesiredHeight = 50
-        val mockResources = ctx.resources
-
-        val spannable = RoadShieldGenerator.styleAndGetRoadShield(
-            mockShieldText,
-            mockDesiredHeight,
-            mockResources
+            mockResources,
+            RouteShieldFactory.buildRouteShield(
+                "https://mapbox_legacy_url.svg",
+                emptyShieldIcon,
+                "https://mapbox_legacy_url"
+            )
```

Comment:
Why using a concrete instance instead of a mock? This way we wouldn't have to expose the `RouteShieldFactory` since it's only needed for tests, right?

Reply:
No this is also needed here
```
 
 ​    @Deprecated( 
 ​        message ​=​ ​"​The method is incapable of rendering Mapbox designed route shields. In cases ​"​ ​+ 
 ​            ​"​where an instruction contains multiple shields, the method may only render 1 shield ​"​ ​+ 
 ​            ​"​instead of multiple shields for that maneuver.​"​, 
 ​        replaceWith ​=​ ​ReplaceWith​(​"​renderManeuverWith(shields)​"​) 
 ​    ) 
 ​    ​fun​ ​renderManeuverShields​(​shieldMap​:​ ​Map​<​String​, ​RoadShield​?>) { 
 ​        shieldMap.forEach { entry ​-> 
 ​            ifNonNull(entry.value) { value ​-> 
 ​                routeShields.add(value.toRouteShield()) 
 ​            } 
 ​        } 
 ​        renderManeuvers() 
 ​    }
```
to map legacy RoadShield to RouteShield.MapboxLegacyShield